### PR TITLE
Add Missing Catalog permissions for `/analyze-location` and `/validate-entity` endpoints

### PR DESCRIPTION
**Pull request by 04kash**
## Hey, I just made a Pull Request!

`POST /analyze-location` and `POST /validate-entity` endpoints are exposed and not protected by the permission framework. Since there may be some schema processing/location resolution when calling these endpoint, they could impact the service if not properly protected. 
Changes made:
- The `analyze-location` endpoint is now protected by the `catalog.location.analyze` permission.
- The `validate-entity` endpoint is now protected by the `catalog.entity.validate` permission.

#### :heavy_check_mark: Checklist

- [X] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [X] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [X] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: .changeset/spicy-files-happen.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+---
+'@backstage/plugin-catalog-backend': patch
+'@backstage/plugin-catalog-node': patch
```
Comment (benjdlambert) [inline .changeset/spicy-files-happen.md:3]:
```suggestion
'@backstage/plugin-catalog-backend': minor
'@backstage/plugin-catalog-node': minor
```

**Review by benjdlambert — CHANGES_REQUESTED** [review]
Think it would be a little unfortunate to bundle these permissions together with the existing ones.

Thinking that `catalog.location.analyze` and `catalog.entity.validate` probably make the most sense here.

File: beps/00010-event-auditor/README.md [hunk 2]
```diff
@@ -115,12 +115,16 @@ export type AuditEventFailureStatus<E = ErrorLike> = {
   errors: E[];
```
Comment (benjdlambert) [inline beps/00010-event-auditor/README.md:115]: Can undo these changes here.

File: package.json [hunk 3]
```diff
@@ -92,6 +92,7 @@
     "jest-haste-map@^29.7.0": "patch:jest-haste-map@npm%3A29.7.0#./.yarn/patches/jest-haste-map-npm-29.7.0-e3be419eff.patch"
   },
   "dependencies": {
+    "@backstage/backend-plugin-api": "workspace:^",
```
Comment (benjdlambert) [inline package.json:95]: Don't think that this should be here :)

File: plugins/catalog-backend/src/service/AuthorizedCatalogProcessingOrchestrator.ts [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+  CatalogProcessingOrchestrator,
+  EntityProcessingRequest,
+  EntityProcessingResult,
+} from '../processing/types';
+import {
+  BackstageCredentials,
+  PermissionsService,
+} from '@backstage/backend-plugin-api';
+
+export class AuthorizedCatalogProcessingOrchestrator
```
Comment (benjdlambert) [inline plugins/catalog-backend/src/service/AuthorizedCatalogProcessingOrchestrator.ts:29]: I think that we should convert this into a `ValidationService` class instead which is called directly by the router to validate with a `CatalogProcessingOrchestrator` instead. 

It feels a little odd to add the `BackstageCredentials` argument to the `Orchestrator` which shouldn't really have any knowledge of that, its designed to be used with the processing loop and not aware of incoming http requests or whatnot.

File: plugins/catalog-backend/src/processing/types.ts [hunk 5]
```diff
@@ -28,6 +29,7 @@ import {
 export type EntityProcessingRequest = {
   entity: Entity;
   state?: JsonObject; // Versions for multiple deployments etc
+  credentials?: BackstageCredentials;
```
Comment (benjdlambert) [inline plugins/catalog-backend/src/processing/types.ts:32]: Think that we can drop this when we move all the permissions to the router instead.

**Review by benjdlambert — CHANGES_REQUESTED** [review]
Hey :wave: thanks for this!

I think that were on the right track here, but there's some high level comments here that I think we need to address before moving forward.

File: plugins/catalog-backend/src/service/AuthorizedCatalogProcessingOrchestrator.ts (hunk 4 above)
Comment (04kash) [reply plugins/catalog-backend/src/service/AuthorizedCatalogProcessingOrchestrator.ts:29]: I chose this approach to maintain consistency with how permissions are handled across the catalog. For example, [`EntitiesCatalog`](https://github.com/backstage/backstage/blob/master/plugins/catalog-backend/src/service/AuthorizedEntitiesCatalog.ts#L111), [`LocationService`](https://github.com/backstage/backstage/blob/master/plugins/catalog-backend/src/service/AuthorizedLocationService.ts#L60), [`RefreshService`](https://github.com/backstage/backstage/blob/master/plugins/catalog-backend/src/service/AuthorizedRefreshService.ts#L44) all have `BackstageCredentials` as an argument. I don't see why the `Orchestrator` would be any different in this regard.

Comment (benjdlambert) [reply plugins/catalog-backend/src/service/AuthorizedCatalogProcessingOrchestrator.ts:29]: Semantically I don't think it makes that much sense. You mentioned also the `LocationService` and the `RefreshService` which is the pattern I'm trying to recommend here too, the `Orchestrator` is used in a lot more places too than just the router. Just feeling that this should be taken care of at the Routing level as much as possible. Let's see how it feels?

Comment (04kash) [reply plugins/catalog-backend/src/service/AuthorizedCatalogProcessingOrchestrator.ts]: @benjdlambert changed it so that permission checking is handled at the router level, lmk what you think.

**Review by benjdlambert — COMMENTED** [review]
Nice! Think that this is looking good! :tada:

**Review by benjdlambert — APPROVED** [review]
